astroquery/cadc/core.py

```diff
@@ -237,6 +238,121 @@ def get_collections(self):
+        fits files with cutouts around the passed in coordinates.
+
+        Parameters
+        ----------
+        coordinates : str or `astropy.coordinates`.
+            Coordinates around which to query
+        radius : str or `astropy.units.Quantity`.
+            The radius of the cone search AND cutout area
+        collection: str, optional
+            Name of the CADC collection to query, optional
+        get_query_payload : bool, optional
+            Just return the dict of HTTP request parameters.
+        show_progress: bool, optional
+            Whether to display a progress bar if the file is downloaded
+            from a remote server.  Default is `True`.
+
+        Returns
+        -------
+        list : A list of `~astropy.io.fits.HDUList` objects
+        """
+        # TODO: Implement threading
+        request_payload = self._args_to_payload(coordinates=coordinates,
+                                                radius=radius,
+                                                collection=collection)
+
+        if get_query_payload:
+            return request_payload
+
+        query_result = self.query_region(coordinates, radius=radius, collection=collection)
+        images_urls = self.get_image_list(query_result, coordinates, radius)
+        readable_objects = [commons.FileContainer(url, encoding='binary',
+                                                  show_progress=show_progress) for url in images_urls]
+
+        return [obj.get_fits() for obj in readable_objects]
+
+    @class_or_instance
+    def get_image_list(self, query_result, coordinates, radius):
+        """
+        Function to map the results of a CADC query into URLs to
+        corresponding data and cutouts that can be later downloaded.
+
+        The function uses the IVOA DataLink Service
+        (http://www.ivoa.net/documents/DataLink/) implemented at the CADC.
+        It works directly with the results produced by Cadc.query_region and
+        Cadc.query_name but in principle it can work with other query
+        results produced with the Cadc query as long as the results
+        contain the 'caomPublisherID' column. This column is part of the
+        caom2.Plane table.
+
+        Parameters
+        ----------
+        query_result : result returned by Cadc.query_region() or
+                    Cadc.query_name(). In general, the result of any
+                    CADC TAP query that contains the 'caomPublisherID' column
+                    can be use here.
+        coordinates : str or `astropy.coordinates`.
+            Coordinates around which to query
+        radius : str or `astropy.units.Quantity`.
+            The radius of the cone search AND cutout area
+
+        Returns
+        -------
+        A list of URLs to data.
+        """
+
+        # TODO: Get access url from publisher ID to find which datalink service to use (i.e. datalink or sc2links)
+        # TODO: Check datalink reponse
+
+        def chunks(l, n):
```

Comment:
Could you use variable names with >1 character?  That's part of our style preference in astroquery, in part so that we can easily search for those variable names.

Reply:
Sure, makes sense :+1: